src/bech32.c

```diff
@@ -78,36 +89,36 @@ static int bech32_encode(char *output, const char *hrp, const uint8_t *data, siz
     for (i = 0; i < 6; ++i) {
         chk = bech32_polymod_step(chk);
     }
-    chk ^= 1;
+    chk ^= bech32_final_constant(enc);
```

Comment:
pass the checksum constant directly to this function and use it here, there is then no need to reason about the failure case in bech32_final_constant().